include/mrdocs/Metadata/NamespaceAlias.hpp

```diff
@@ -0,0 +1,46 @@
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// Copyright (c) 2024 Fernando Pelliccioni (fpelliccioni@gmail.com)
+//
+// Official repository: https://github.com/cppalliance/mrdocs
+//
+
+#ifndef MRDOCS_API_METADATA_NAMESPACEALIAS_HPP
+#define MRDOCS_API_METADATA_NAMESPACEALIAS_HPP
+
+#include <mrdocs/Platform.hpp>
+#include <mrdocs/Metadata/Info.hpp>
+#include <mrdocs/Metadata/Source.hpp>
+#include <mrdocs/Metadata/Type.hpp>
+#include <utility>
+
+namespace clang {
+namespace mrdocs {
+
+/** Info for namespace aliases.
+*/
+struct AliasInfo
+    : IsInfo<InfoKind::Alias>
+    , SourceInfo
+{
+    /** Aliased symbol. */
+    SymbolID AliasedSymbol = SymbolID::invalid;
+
+    /** The qualifier for the alias. */
+    std::unique_ptr<NameInfo> Qualifier;
```

Comment:
For namespace aliases, the `NameInfo` shouldn't store just the qualifier, but the full qualified name (including the terminal name). There should be an overload for building a `NameInfo` from a `NamedDecl`, so you should use that